docs/clean/user_guide/writing_a_clean/device_recovery.rst

```diff
@@ -0,0 +1,163 @@
+.. _clean_doc_device_recovery:
+Device Recovery
+===============
+
+Before using your new `Clean` it is important to learn about `Device Recovery` as it is the **second most important** aspect to
+pyATS Clean.
+
+.. topic:: Topics covered in this section are:
+
+    * `The Methodology Behind Device Recovery`_
+    * `How Device Recovery Works`_
+    * `How to Enable Device Recovery`_
+
+The Methodology Behind Device Recovery
+--------------------------------------
+
+pyATS Clean provides a jack-of-all-trades for **recovering devices** that fail due to device malfunctions or if the clean
+process itself fails to execute (see below for details). `Device Recovery` can clear the line of a device from the terminal server
+and use a powercycler to restart the device. After restarting the device, `Device Recovery` will load the **golden image**
+and optionally apply a configuration onto the device.
+
+The major advantage the device recovery feature provides is—you can **recover a device and reset it back to a useable
+state** regardless of the current state of the device.
+
+How Device Recovery Works
+-------------------------
+
+Device Recovery is invoked after every stage and it will determine whether to continue or fail clean.
+
+.. topic:: This may depend on:
+
+    * The result of the last executed stage.
+    * The connectivity to the device.
+    * If the last stage ran was the `connect` stage.
+
+**The following diagram explains, in detail, the flow of Device Recovery.**
+
+.. image:: ../../assets/pyATS_clean_device_recovery_diagram.png
```

Comment:
text is blury, can you get a larger picture

Reply:
switched to svg image